internal/scheduler/job/repository_run.go

```diff
@@ -86,33 +86,37 @@ func (r *Repository) UpdateProgress(ctx context.Context, runId string, completed
 
 	run := allocRun()
 	run.PrivateId = runId
-	if err := r.reader.LookupById(ctx, run); err != nil {
-		if errors.IsNotFoundError(err) {
-			return nil, errors.Wrap(err, op, errors.WithMsg(fmt.Sprintf("run %q does not exist", runId)))
-		}
-		return nil, errors.Wrap(err, op)
-	}
 	_, err = r.writer.DoTx(ctx, db.StdRetryCnt, db.ExpBackoff{},
 		func(r db.Reader, w db.Writer) error {
 			rows, err := r.Query(ctx, updateProgressQuery, []interface{}{completed, total, runId})
 			if err != nil {
 				return errors.Wrap(err, op)
 			}
-			defer rows.Close()
 
 			var rowCnt int
 			for rows.Next() {
 				if rowCnt > 0 {
+					_ = rows.Close()
```

Comment:
Why the switch away from `defer rows.Close()`?

Reply:
I need to close the rows for the reader to do `LookupById` below so I could not wait for defer